### PR TITLE
improve explanation of exception when two sprints with the same name are found

### PR DESCRIPTION
**Pull request by studioj**

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -4516,13 +4516,27 @@ def sprints(
             self.AGILE_BASE_URL,
         )
 
-    def sprints_by_name(self, id, extended=False):
+    def sprints_by_name(self, id, extended=False, state=None):
```
Comment (adehad) [inline jira/client.py:4519]: shall we also add typehints here?

```suggestion
    def sprints_by_name(self, id: Union[str,int], extended: bool = False, state: str = None) -> Dict[str, Dict[str, Any]]:
```

the output type is a bit weird for this function (ie. could spit out the sprint object itself?)... but probably beyond this PR

File: jira/client.py [hunk 2]
```diff
@@ -4516,13 +4516,27 @@ def sprints(
...
+        """Get a dictionary of sprint Resources where the name of the sprint is the key.
+
+        Args:
+            board_id (int): the board to get sprints from
+            extended (bool): Deprecated.
+            state (str): Filters results to sprints in specified states. Valid values: `future`, `active`, `closed`.
+              You can define multiple states separated by commas
```
Comment (adehad) [inline jira/client.py:4526]: is this case sensitive?

File: tests/resources/test_sprint.py [hunk 3]
```diff
@@ -97,3 +100,11 @@ def test_move_issue_to_backlog(self):
             # THEN: There is no longer the sprint assigned
             updated_issue_1 = self.jira.issue(self.issue_1)
             assert updated_issue_1.get_field(self._sprint_customfield()) is None
+
+    def test_two_sprints_with_the_same_name_raise_a_jirra_error_when_sprints_by_name_is_called(
```
Comment (adehad) [inline tests/resources/test_sprint.py:104]:
```suggestion
    def test_two_sprints_with_the_same_name_raise_a_jira_error_when_sprints_by_name_is_called(
```

File: jira/client.py [hunk 4]
```diff
@@ -4516,13 +4516,27 @@ def sprints(
...
+
+        Returns:
+            dict: dictionary of sprints with the sprint name as key
```
Comment (adehad) [inline jira/client.py:4529]:
```suggestion
            Dict[str, Dict[str, Any]]: dictionary of sprints with the sprint name as key
```

**Review by adehad — CHANGES_REQUESTED** [review]

File: jira/client.py (hunk 2 above)
Comment (studioj) [reply jira/client.py:4526]: no clue didnt validate this just copied from sprints()

Comment (adehad) [reply jira/client.py:4528]: beyond this PR then

**Pull request merged**